### ISSUE TITLE
`solid_selection` in `execute_pipeline` seems broken

### ISSUE THREAD

**Issue by mansenfranzen** [dagster-io/dagster#3186]
Hi dagster-people,

great framework- I really enjoy using it! I have a problem selecting solids while using `execute_pipeline`. 

### Minimal reproducible example

```python
from dagster import solid, pipeline

@solid
def step1(_, param: str) -> str:
    return param + "Foo"

@solid
def step2(_, param: str) -> str:
    return param + "Bar"

@solid
def step3(_, foo: str, bar: str) -> str:
    return foo + bar

@solid
def step4(_, foobar: str) -> int:
    return len(foobar)

@solid
def random_step(_, param: str) -> str:
    return param

@pipeline
def pipe():
    random_step()
    
    s1 = step1()
    s2 = step2()
    s3 = step3(s1, s2)
    s4 = step4(s3)

test_conf = {
    "solids": {
        "step1": {"inputs": {"param": "asd"}},
        "step2": {"inputs": {"param": "bsd"}},
        "random_step": {"inputs": {"param": "foobar"}},
    }
}
```

Here is the corresponding graph:

![image](https://user-images.githubusercontent.com/18086180/98119857-7a011980-1ead-11eb-970b-4d7ec782317d.png)

### Issue
Now, for example, I may just want to run one step 1, or step3 with all upstream dependencies, however both fail with  `DagsterInvalidConfigError`.

#### Only step1

```python
execute_pipeline(pipe, run_config=test_conf, solid_selection=["step1"])

---------------------------------------------------------------------------
DagsterInvalidConfigError                 Traceback (most recent call last)
<ipython-input-53-7a669fe5181a> in <module>
----> 1 execute_pipeline(pipe, run_config=test_conf, solid_selection=["step1"])

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\execution\api.py in execute_pipeline(pipeline, run_config, mode, preset, tags, solid_selection, instance, raise_on_error)
    330             tags=tags,
    331             solid_selection=solid_selection,
--> 332             raise_on_error=raise_on_error,
    333         )
    334 

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\telemetry.py in wrap(*args, **kwargs)
     88         start_time = datetime.datetime.now()
     89         log_action(instance=instance, action=f.__name__ + "_started", client_time=start_time)
---> 90         result = f(*args, **kwargs)
     91         end_time = datetime.datetime.now()
     92         log_action(

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\execution\api.py in _logged_execute_pipeline(pipeline, instance, run_config, mode, preset, tags, solid_selection, raise_on_error)
    370         solid_selection=solid_selection,
    371         solids_to_execute=solids_to_execute,
--> 372         tags=tags,
    373     )
    374 

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\instance\__init__.py in create_run_for_pipeline(self, pipeline_def, execution_plan, run_id, run_config, mode, solids_to_execute, step_keys_to_execute, status, tags, root_run_id, parent_run_id, solid_selection)
    581 
    582         full_execution_plan = execution_plan or create_execution_plan(
--> 583             pipeline_def, run_config=run_config, mode=mode,
    584         )
    585         check.invariant(

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\execution\api.py in create_execution_plan(pipeline, run_config, mode, step_keys_to_execute)
    639     check.opt_list_param(step_keys_to_execute, "step_keys_to_execute", of_type=str)
    640 
--> 641     environment_config = EnvironmentConfig.build(pipeline_def, run_config, mode=mode)
    642 
    643     return ExecutionPlan.build(

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\system_config\objects.py in build(pipeline_def, run_config, mode)
     99                 "Error in config for pipeline {}".format(pipeline_def.name),
    100                 config_evr.errors,
--> 101                 run_config,
    102             )
    103 

DagsterInvalidConfigError: Error in config for pipeline pipe
    Error 1: Fields "['random_step', 'step2']" are not defined at path root:solids. Available fields: "['step1']."
```
### Step3 and all upstream dependencies

```python
execute_pipeline(pipe, run_config=test_conf, solid_selection=["*step3"])

---------------------------------------------------------------------------
DagsterInvalidConfigError                 Traceback (most recent call last)
<ipython-input-54-b4c4842a059c> in <module>
----> 1 execute_pipeline(pipe, run_config=test_conf, solid_selection=["*step3"])

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\execution\api.py in execute_pipeline(pipeline, run_config, mode, preset, tags, solid_selection, instance, raise_on_error)
    330             tags=tags,
    331             solid_selection=solid_selection,
--> 332             raise_on_error=raise_on_error,
    333         )
    334 

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\telemetry.py in wrap(*args, **kwargs)
     88         start_time = datetime.datetime.now()
     89         log_action(instance=instance, action=f.__name__ + "_started", client_time=start_time)
---> 90         result = f(*args, **kwargs)
     91         end_time = datetime.datetime.now()
     92         log_action(

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\execution\api.py in _logged_execute_pipeline(pipeline, instance, run_config, mode, preset, tags, solid_selection, raise_on_error)
    370         solid_selection=solid_selection,
    371         solids_to_execute=solids_to_execute,
--> 372         tags=tags,
    373     )
    374 

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\instance\__init__.py in create_run_for_pipeline(self, pipeline_def, execution_plan, run_id, run_config, mode, solids_to_execute, step_keys_to_execute, status, tags, root_run_id, parent_run_id, solid_selection)
    581 
    582         full_execution_plan = execution_plan or create_execution_plan(
--> 583             pipeline_def, run_config=run_config, mode=mode,
    584         )
    585         check.invariant(

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\execution\api.py in create_execution_plan(pipeline, run_config, mode, step_keys_to_execute)
    639     check.opt_list_param(step_keys_to_execute, "step_keys_to_execute", of_type=str)
    640 
--> 641     environment_config = EnvironmentConfig.build(pipeline_def, run_config, mode=mode)
    642 
    643     return ExecutionPlan.build(

~\.conda\envs\env_ftfr\lib\site-packages\dagster\core\system_config\objects.py in build(pipeline_def, run_config, mode)
     99                 "Error in config for pipeline {}".format(pipeline_def.name),
    100                 config_evr.errors,
--> 101                 run_config,
    102             )
    103 

DagsterInvalidConfigError: Error in config for pipeline pipe
    Error 1: Undefined field "random_step" at path root:solids. Expected: "{ step1: { inputs: { param: (String | { json: { path: String } pickle: { path: String } value: String }) } outputs?: [{ result?: { json: { path: String } pickle: { path: String } } }] } step2: { inputs: { param: (String | { json: { path: String } pickle: { path: String } value: String }) } outputs?: [{ result?: { json: { path: String } pickle: { path: String } } }] } step3?: { outputs?: [{ result?: { json: { path: String } pickle: { path: String } } }] } }".
```

Perhaps I'm missing here something but I couldn't find anything more in the documentation. Btw I'm using dagster 0.9.16 on python 3.7 windows.

**Comment by alangenfeld** [comment]
So the errors you are getting here are from your run config `test_conf` containing config for all the solids despite only executing a subset. You can move forward by altering the config passed in to `execute_pipeline` to only contain config for solids that exist in the defined solid selection.

This is a known usability problem and there is a diff out for fixing it. The plan is to gracefully ignoring the config for solids that exist in the whole pipeline but not needed in the current solid selection.

**Comment by mansenfranzen** [comment]
Thanks for the answer. It is understandable why it works that way but it really shouldn't for usability reasons as you pointed out. For now, I guess some hint in the documentation would be nice to have.

**Comment by alangenfeld** [comment]
fix went out in `0.9.19`
